### ISSUE TITLE
Quicksilver 1.5.1(build 401C) can't startup on macOS Sierra

### ISSUE THREAD

**Issue by galeo** [quicksilver/Quicksilver#2299]
Thanks for all your hard work for the latest release of QS. However, after upgrading to version 1.5.1, it can't startup. 

![screen shot 2016-12-28 at 8 00 13 pm](https://cloud.githubusercontent.com/assets/1025559/21521518/b7df70f0-cd38-11e6-91c6-f6ff4569b47a.png)

I have to turn back to version 1.4.2 :( 
Any further information needed?

**Comment by skurfer** [comment]
Hello. Can you try running 1.5.1 again, select it in Activity Monitor, and choose “Spindump” from the gear menu? The file might be a little large to paste here, but you can upload it somewhere and share the link or create a [Gist](https://gist.github.com/). Thanks.

**Comment by galeo** [comment]
Hi, @skurfer 

Sorry for the late reply, here is the gist of the "Spindump" log: https://gist.github.com/galeo/0e96e48f4c6e216d75abc445029007e8

Thanks.

**Comment by skurfer** [comment]
Try this test build. It *might* fix the issue, but more likely just moves it to another spot. However, I’ve added some logging where I think the problem is, so hopefully we can narrow it down.

https://cl.ly/2e02333H1O33

If it still hangs, look in the Console app for messages from Quicksilver starting with “UT tag class” or “UTI for type string”. I’m hoping the last message it logs before it freezes will contain the string that’s confusing it.

**Comment by galeo** [comment]
I've just tried the test build version and It works well. It seems that the hang problem has been fixed.

Thanks so much.

**Comment by skurfer** [comment]
OK. I’ll add the change I made to the app. Feel free to keep using the test build, but be aware that it might be a bit noisy in your logs.

**Comment by galeo** [comment]
I'm sorry, the hang problem occurred again.

Here is the last logs:
```
Jan  4 19:37:21 galeo-macbook Quicksilver[2405]: UTI for type string: QSObjectName
Jan  4 19:37:21 galeo-macbook Quicksilver[2405]: UT tag class: com.apple.ostype
Jan  4 19:37:22 galeo-macbook Quicksilver[2405]: UT tag class: public.filename-extension
Jan  4 19:37:22 galeo-macbook Quicksilver[2405]: UT tag class: public.mime-type
Jan  4 19:37:22 galeo-macbook Quicksilver[2405]: UT tag class: com.apple.nspboard-type
Jan  4 19:37:22 galeo-macbook Quicksilver[2405]: UTI for type string: QSLineReferenceType
Jan  4 19:37:22 galeo-macbook Quicksilver[2405]: UT tag class: com.apple.ostype
Jan  4 19:37:22 galeo-macbook Quicksilver[2405]: UT tag class: public.filename-extension
Jan  4 19:37:22 galeo-macbook Quicksilver[2405]: UT tag class: public.mime-type
... ...
Jan  4 19:37:45 galeo-macbook Quicksilver[2405]: UTI for type string: Apple URL pasteboard type
Jan  4 19:37:45 galeo-macbook Quicksilver[2405]: UT tag class: com.apple.ostype
Jan  4 19:37:45 galeo-macbook Quicksilver[2405]: UT tag class: public.filename-extension
Jan  4 19:37:45 galeo-macbook Quicksilver[2405]: UT tag class: public.mime-type
```

**Comment by skurfer** [comment]
Give this a try. https://cl.ly/1O043N0o0W0W

It should be just a small executable. Save it somewhere and run it in Terminal. All it does is attempt to look up the MIME type for “Apple URL pasteboard type”, which seems to be where you’re getting stuck based on the above. I’m curious to see if the command hangs on your system.

On mine, it just outputs the result:

    2017-01-11 08:31:48.193 types[29581:1391069] dyn.agq80c6durvwwa7pwruuha2pxsvw0e55bsmwca7d3sbwu

**Comment by galeo** [comment]
It outputs on mine, just the same with yours:
```
2017-01-11 21:51:01.393 types[4327:186900] dyn.agq80c6durvwwa7pwruuha2pxsvw0e55bsmwca7d3sbwu
```

**Comment by skurfer** [comment]
I can’t say that I’m surprised, but I’m also not sure what to make of it. Based on the logs and the spin dump, that command should be doing the same thing QS gets stuck on. Of course, QS is trying to do it for multiple types rapid-fire.

Maybe your Launch Services database has some corruption? You can rebuild it by running

    /System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Support/lsregister -kill -r -domain local -domain system -domain user

Before you do something that drastic though, we should consider the possibility that Quicksilver’s getting hung in a different spot now. Feel free to try another spindump and we can check.

**Comment by galeo** [comment]
I rebuilt the Launch Services database and tried to launch the test build version of QS that given by you. It could startup the first time after reinstallation. However, when quit it and try to launch it again, it still hangs.

Here is the newer spindump gist: https://gist.github.com/galeo/335fb077d3ece4d98f43d65b9a92ce94

Thanks a lot.

**Comment by tiennou** [comment]
Can you try to remove the cached indexes ? They are located under ~/Library/Caches/Quicksilver/Indexes.

**Comment by galeo** [comment]
Hi @tiennou,

It can't solve the problem by removing the cached indexes, the "QS" icon hangs at the center of screen.

**Comment by tiennou** [comment]
Yeah, that was a shot in the dark, assuming that since it's stuck doing lots of UTI resolutions while loading indexes there could be something broken in there.

IIUC, after you reset the LS database, QS runs once, then on a subsequent run it hangs ?

@skurfer Could it be that manipulating UTIs cause us to corrupt our own "private" database in some way ? So the first time it runs, we declare all those UTIs, but the next time around since they already exist we get stuck on conformance tests ? It's a complete shot in the dark (again), but it's the first time I get to see LS underlying behavior so closely 😉 .

**Comment by skurfer** [comment]
Give this test build a try and let me know if it’s any better. The version should show as “1.5.2:mine”.

https://cl.ly/0c3v0M031t3L

**Comment by galeo** [comment]
It immediately crashed after launching. 

Here is the crash report:  https://gist.github.com/galeo/dad8bfdf8d944c7b05f91c2d0ef0af27

**Comment by skurfer** [comment]
Sorry. A debugging setting snuck into the release build. See if this is any better.

https://cl.ly/420r2R0z3R3d

**Comment by galeo** [comment]
Sorry for the late reply. The test build version(1.5.2:mine) runs once, then it still hangs after relaunching.

**Comment by skurfer** [comment]
This changes some of the behavior when reloading the catalog from disk after a relaunch.

https://cl.ly/0N3W180S2G45

Let me know if it’s any better.

**Comment by galeo** [comment]
It works OK now except that it's still not responding (high CPU consumption) for 10 seconds or so.

Thanks.

**Comment by skurfer** [comment]
I was able to reproduce (and maybe fix) some of these delays. Try this: https://cl.ly/011O462y3r3C
